service/distros.go

```diff
@@ -122,6 +122,13 @@ func (uis *UIServer) modifyDistro(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure docker password wasn't auto-filled from form
+	if newDistro.Provider != evergreen.ProviderNameDocker && newDistro.Provider != evergreen.ProviderNameDockerMock {
```

Comment:
Is it possible to make this more generalized so that we delete all settings that aren't relevant to the type?

Reply:
I thought about that, but not without roundtripping into the right settings object which feels like overkill--I think this problem will go away with the distro ticket mentioned above anyway (or I could stop using that password form field thing) so this is more a temporary fix.